text/0103-introduce-core-index-commitment.md

```diff
@@ -0,0 +1,252 @@
+	pub fn signature(&self) -> Option<CollatorSignature>;
+
+	/// Returns the core index of the descriptor. Returns `None` if the descriptor is at version 1.
+	pub fn core_index(&self) -> Option<CoreIndex>;
+
+	/// Returns the session index of the descriptor. Returns `None` if the descriptor is at version 1.
+	pub fn session_index(&self) -> Option<SessionIndex>;
+
+	/// ...
+}
+
+```
+
+A manual decode `Decode`  implementation is required to account for version detection and constructing the appropriate variant.
+
+### Parachain block validation
+
+#### Node
+
+Backers will make use of the core index information to validate the blocks during backing and reject blocks if:
+- the `core_index` in descriptor does not match the one determined by the `UMPSignal::SelectCore` message
+- the `core_index` in the descriptor does not match the core the backing group is assigned to
+- the `session_index` is not equal to the session of the `relay_parent` in the descriptor
+
+If core index (and session index) information is not available (backers got an old candidate receipt), there will be no changes compared to current behaviour.
+
+#### Runtime
+
+The runtime will also perform the above checks and reject invalid candidates.
+
+## Drawbacks
+
+The only drawback is that further additions to the descriptor are limited to the amount of remaining unused space.
+
+## Testing, Security, and Privacy
+
+Standard testing (unit tests, CI zombienet tests) for functionality and mandatory secuirty audit to ensure the implementation does not introduce any new security issues.
+
+Backwards compatibility of the implementation will be tested on testnets (Versi and Westend).
+
+There is no impact on privacy.
+
+## Performance
+
+The expectation is that performance impact is negligible for sending and processing the UMP message has negligible performance impact in the runtime as well as on the node side.
+
+## Ergonomics
+
+It is mandatory for elastic parachains to switch to the new receipt format. It is optional but desired that all parachains
+switch to the new receipts for providing the session index for disputes.
+
+Once this RFC is implemented the parachain runtime and node must not require any manual changes to use it, except if the parachain wants to change the `ClaimQueueOffset` that is used to determine the core index. 
+
+## Compatibility
+
+The proposed changes are backwards compatible in general, but additional care must be taken by waiting for enough validators to upgrade before the validators and runtime start accepting the new candidate receipts.
+
+### Runtime
+
+The first step is to remove collator signature checking logic in the runtime, but keep the node side collator signature 
+checks. 
+
+The runtime must be upgraded to the new primitives before any collator or node are allowed to use the new candidate receipts format. 
+
+### Validators
+
+To ensure a smooth launch, a new node feature is required. 
```

Comment:
the runtime will also reject candidates that use v2 if the node feature is not set

Reply:
Yes, but that is temporary implementation detail which is not worth mentioning.

Reply:
why is this temporary?
Personally I think it's worth mentioning